### ISSUE TITLE
[Bug] CrossEntropyLoss function doesn't work for training.

### ISSUE THREAD

**Issue by echuraev** [mlc-ai/relax#127]
Hello @SiriusNEO, @Ubospica!

Thank you for your hard work on adding training to TVM. It is awesome work!
My student @ndemashov is using this functionality at his research and he found an issue that it wasn't possible to use `CrossEntropyLoss` function for training.

Probably you could correct us if we use `CrossEntropyLoss` function in incorrect way. The commit with the failing test and some other minor changes is presented at this commit: https://github.com/echuraev/relax/commit/8f7efa64523333ac5349d013bfe319e4f2472d01

Please take a look at the commit message of this commit. It contains detailed description of all issues which were occurred while our experiments.

From my point of view, it is necessary to implement gradient function for `nll_loss` function. If it is true (I believe this true), I think it might be a good task for @ndemashov and he can fix that. It will help him to dive deeper to training in TVM. What do you think about that?
Do you have any ideas about the issue with different data types for the gradient function and how it should be fixed? I didn't spend much time on looking the formula of gradient for nll_loss function, so if you have any links, could you please share them with us?

### Expected behavior

Training should work with `CrossEnropyLoss` function

### Actual behavior

It doesn't work.

### Environment

Ubuntu Linux on one machine and WSL on another. The latest state of `mlc-ai/relax` branch.

### Steps to reproduce

Test was added at this commit: https://github.com/echuraev/relax/commit/8f7efa64523333ac5349d013bfe319e4f2472d01

**Comment by echuraev** [comment]
Close this issue because found this PR: https://github.com/mlc-ai/relax/pull/114

**Comment by SiriusNEO** [comment]
Yes, the problem is that we have not implemented the gradient of `nll_loss` because of its complexity. PR https://github.com/mlc-ai/relax/pull/114 will solve this problem. And for now we can use `log_softmax + cross_entropy_with_logits` (both have gradients) if necessary.
